File: myself/src/main/java/com/xxd/myself/intdef/MyAnnotation.java
package com.xxd.myself.intdef;

import androidx.annotation.IntDef;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * author : xxd
 * date   : 2023/12/3
 * desc   :
 */

@IntDef({DomainA.Type1,DomainA.Type2,DomainA.Type3})
@Target({ElementType.TYPE,ElementType.FIELD,ElementType.PARAMETER})
@Retention(RetentionPolicy.SOURCE)
public @interface MyAnnotation {
}

File: myself/src/main/java/com/xxd/myself/intdef/kt/MyAnnotation.kt
package com.xxd.myself.intdef.kt

import androidx.annotation.IntDef

/**
 *    author : xxd
 *    date   : 2023/12/3
 *    desc   :
 */
@IntDef(DomainB.type1, DomainB.type2, DomainB.type3)
@Retention(AnnotationRetention.SOURCE)
@Target(AnnotationTarget.VALUE_PARAMETER)
annotation class MyAnnotation {
}